adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -88,23 +88,41 @@ - (void)renderNativeAdForAdConfiguration:
                  strongSelf->_nativeAd = nativeAd;
                  strongSelf->_nativeAd.delegate = strongSelf;
                  strongSelf->_nativeAd.rootViewController = adConfiguration.topViewController;
-
-                 if (strongSelf->_loadCompletionHandler) {
-                   id<GADMediationNativeAdEventDelegate> delegate =
-                       strongSelf->_loadCompletionHandler(strongSelf, nil);
-                   strongSelf->_delegate = delegate;
-                 }
+      
+                 [strongSelf loadRequireData];
                }];
 }
 
-#pragma mark - GADMediationNativeAd
-
-- (GADNativeAdImage *)icon {
-  if (!_icon) {
-    if (_nativeAd.data.icon && _nativeAd.data.icon.imageURL != nil) {
-      _icon = [self imageWithUrlString:_nativeAd.data.icon.imageURL];
+- (void)loadRequireData {
+    NSString *urlString = _nativeAd.data.icon.imageURL;
+    if (!urlString.length) {
+        if (_loadCompletionHandler) _delegate = _loadCompletionHandler(self,nil);
+        return;
     }
-  }
+    NSURL *url = [NSURL URLWithString:urlString];
+    if (!url) {
+        if (_loadCompletionHandler) _delegate = _loadCompletionHandler(self,nil);
+        return;
+    }
+    dispatch_async(dispatch_get_global_queue(0, 0), ^{
+        NSData *data = [NSData dataWithContentsOfURL:url];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (!self) {
+                return;
+            }
+            GADNativeAdImage *image = [[GADNativeAdImage alloc] initWithImage:[UIImage imageWithData:data]];
+            self->_icon = image;
+            if (self->_loadCompletionHandler) {
+              id<GADMediationNativeAdEventDelegate> delegate =
+                  self->_loadCompletionHandler(self, nil);
+              self->_delegate = delegate;
+            }
+        });
+    });
+}
+
+#pragma mark - GADMediationNativeAd
+- (nullable GADNativeAdImage *)icon {
   return _icon;
 }
 
```

Comment:
For all the assets, can you add a `nullable` annotation just like what you did for `icon`?

Reply:
ok~